developing-with-nats/tutorials/jwt.md

```diff
@@ -0,0 +1,1540 @@
+[1] 94745
+11:50:41 Subscribing on >
+>nats -s nats://localhost:4222 pub --nkey=b.nk "foo" "nkey"
+11:56:30 [#1] Received on "foo"
+nkey
+
+11:56:30 Published 4 bytes to "foo"
+>
+```
+
+When the nats-server was started with `-V` tracing, you can see the signature in the `CONNECT` message (formatting added manually).
+
+```text
+[95184] 2020/10/26 12:15:44.350577 [TRC] [::1]:55551 - cid:2 - <<- [CONNECT {
+    "echo": true,
+    "headers": true,
+    "lang": "go",
+    "name": "NATS CLI",
+    "nkey": "UC435ZYS52HF72E2VMQF4GO6CUJOCHDUUPEBU7XDXW5AQLIC6JZ46PO5",
+    "no_responders": true,
+    "pedantic": false,
+    "protocol": 1,
+    "sig": "lopzgs98JBQYyRdw1zT_BoBpSFRDCfTvT4le5MYSKrt0IqGWZ2OXhPW1J_zo2_sBod8XaWgQc9oWohWBN0NdDg",
+    "tls_required": false,
+    "verbose": false,
+    "version": "1.11.0"
+}]
+```
+
+On connect, clients are instantly sent the nonce to sign as part of the `INFO` message (formatting added manually). 
+Since `telnet` will not authenticate, the server closes the connection after hitting the [authorization](https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro#authorization-map) timeout.
+
+```text
+> telnet localhost 4222
+Trying ::1...
+Connected to localhost.
+Escape character is '^]'.
+INFO {
+    "auth_required": true,
+    "client_id": 3,
+    "client_ip": "::1",
+    "go": "go1.14.1",
+    "headers": true,
+    "host": "0.0.0.0",
+    "max_payload": 1048576,
+    "nonce": "-QPTE1Jsk8kI3rE",
+    "port": 4222,
+    "proto": 1,
+    "server_id": "NBSHIXACRHUODC4FY2Z3OYXSZSRUBRH6VWIKQNGVPKOTA7H4YTXWJRTO",
+    "server_name": "NBSHIXACRHUODC4FY2Z3OYXSZSRUBRH6VWIKQNGVPKOTA7H4YTXWJRTO",
+    "version": "2.2.0-beta.26"
+}
+-ERR 'Authentication Timeout'
+Connection closed by foreign host.
+```
+
+#### Key Takeaways
+
+* NKEYS are a secure way to authenticate clients
+* Secret private key is never exchanged, only evidence of possession is
+* The public key still needs to be configured
+
+## JSON Web Tokens (JWT) 
+
+### Motivation for JWT
+
+In a large organization the centralized config approach can lead to less flexibility/resistance to change when controlled by one entity.
+Alternatively, instead of operating one infrastructure, it can be deployed more often (say per team) thus making import/export relationships harder as they have to bridge separate systems.
+In order to make accounts truly powerful, they should ideally be configured separately from the infrastructure, only constraint by limits.
+Similar for user. An account contains user but this relationship could be a reference as well, such that alterations to user do not alter the account. 
+User of the same account should be able to connect from anywhere in the same infrastructure and be able to exchange messages as long as they are in the same authentication domain.
+
+#### Key Takeaways
+
+* Break up config into separate artifacts manageable by different entities
+* Separate accounts from config and user from accounts
```

Comment:
Management of Accounts, Configuration, and Users are separated.

Reply:
Add that as separate bullet point?

Reply:
Or could replace, up to you...

Reply:
replaced